### PR TITLE
Rails engine and decidim-bulletin_board.js as asset

### PR DESCRIPTION
**Pull request by agustibr**
This PR changes the gem to be a Rails Engine, also builds the js in the gem assets directory.

### CODE REVIEW

File: decidim-bulletin_board-js/webpack.common.js [hunk 1]
```diff
@@ -1,18 +1,16 @@
...
 
-module.exports = (env) => ({
-  mode: env.development ? "development" : "production",
+module.exports = {
   entry: "./src/index.js",
   output: {
-    path: path.resolve(__dirname, "dist"),
-    filename: "[name].js",
+    path: path.resolve(__dirname, "..", "decidim-bulletin_board-ruby", "app", "assets","javascripts", "decidim", "bulletin_board"),
+    filename: "decidim-bulletin_board-s.js",
```
Comment (beagleknight) [inline decidim-bulletin_board-js/webpack.common.js]: Is the `-s` a typo?

**Review by beagleknight — APPROVED** [review]
It looks great! 🤩 I am just worried about forgetting to run the build command before releasing the gem but maybe we can just add a CI step on the `decidim-bulletin_board-js` folder or something like that.

File: decidim-bulletin_board-js/webpack.common.js (hunk 1 above)
Comment (agustibr) [reply decidim-bulletin_board-js/webpack.common.js]: Yes, it's fixed now 😁

**Review by leio10 — APPROVED** [review]
🤩 Great work!!

**Review by slickepinne — APPROVED** [review]
Awesome!! 🎉

**Pull request merged**